packages/@aws-cdk/aws-stepfunctions-tasks/lib/codebuild/start-build.ts

```diff
@@ -0,0 +1,734 @@
+   * The CMK key encrypts the build output artifacts.
+   *
+   * @default No override
+   */
+  readonly encryptionKeyOverride?: String;
+  /**
+   * A unique, case sensitive identifier you provide to ensure the idempotency of the StartBuild request.
+   * The token is included in the StartBuild request and is valid for 5 minutes.
+   * If you repeat the StartBuild request with the same token, but change a parameter,
+   * AWS CodeBuild returns a parameter mismatch error.
+   *
+   * @default - None
+   */
+  readonly idempotencyToken?: String;
+  /**
+   *  Log settings for this build that override the log settings defined in the build project.
+   *
+   * @default No override
+   */
+  readonly logsConfigOverride?: LogsConfig;
+  /**
+   *  The credentials for access to a private registry.
+   *
+   * @default No override
+   */
+  readonly registryCredentialOverride?: RegistryCredential;
+  /**
+   * The type of credentials AWS CodeBuild uses to pull images in your build.
+   *
+   * @default No override
+   */
+  readonly imagePullCredentialsTypeOverride?: codebuild.ImagePullPrincipalType;
+  /**
+   * Specifies if session debugging is enabled for this build
+   *
+   * @default No override
+   */
+  readonly debugSessionEnabled?: boolean;
+  /**
+   * Enable this flag to override privileged mode in the build project.
+   *
+   * @default No override
+   */
+  readonly privilegedModeOverride?: boolean;
+}
+
+/**
+ * Start a CodeBuild Build as a task
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/connect-codebuild.html
+ */
+export class CodeBuildStartBuild extends sfn.TaskStateBase {
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+    sfn.IntegrationPattern.RUN_JOB,
+  ];
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: cdk.Construct, id: string, private readonly props: CodeBuildStartBuildProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+
+    validatePatternSupported(this.integrationPattern, CodeBuildStartBuild.SUPPORTED_INTEGRATION_PATTERNS);
+
+    this.taskMetrics = {
+      metricPrefixSingular: 'CodeBuildProject',
+      metricPrefixPlural: 'CodeBuildProjects',
+      metricDimensions: {
+        ProjectArn: this.props.project.projectArn,
+      },
+    };
+
+    this.taskPolicies = this.configurePolicyStatements();
+  }
+
+  private configurePolicyStatements(): iam.PolicyStatement[] {
+    let policyStatements = [
+      new iam.PolicyStatement({
+        resources: [this.props.project.projectArn],
+        actions: [
+          'codebuild:StartBuild',
+          'codebuild:StopBuild',
+          'codebuild:BatchGetBuilds',
+          'codebuild:BatchGetReports',
```

Comment:
does `StartBuild` need these permissions? when would they be used?

Reply:
Good point. I will remove them.